core/lib/makeswift/components/customer-group-slot/customer-group-slot.makeswift.tsx

```diff
@@ -0,0 +1,123 @@
+import { Checkbox, Combobox, Group, List, Slot, Style } from '@makeswift/runtime/controls';
+import { ReactNode } from 'react';
+import useSWR from 'swr';
+
+import { CustomerGroupType, CustomersGroup } from '~/app/api/customer/groups/route';
+import { GetCustomerGroupResponse } from '~/client/queries/get-customer';
+import { runtime } from '~/lib/makeswift/runtime';
+
+async function fetchData(): Promise<CustomerGroupType> {
+  const response = await fetch('/api/customer/groups');
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
```

Comment:
I would only throw an error here if we expect each caller to handle the actual error handling. Is that the intent? The one caller I see below uses try/catch. As long as the expectation is that the caller should handle it, this is ok.